include/grpcpp/ext/gcp_observability.h

```diff
@@ -17,25 +17,91 @@
 #ifndef GRPCPP_EXT_GCP_OBSERVABILITY_H
 #define GRPCPP_EXT_GCP_OBSERVABILITY_H
 
+#include <grpc/support/port_platform.h>
+
 #include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include <grpcpp/impl/grpc_library.h>
 
 namespace grpc {
-namespace experimental {
 
-// Initialize GCP Observability for gRPC.
-// This should be called before any other gRPC operations like creating a
-// channel, server, credentials etc.
-// The most common usage would call this at the top (or near the top) in main().
-// As an implementation detail, this properly initializes the OpenCensus stats
-// and tracing plugin, so applications do not need to perform any additional
-// gRPC C++ OpenCensus setup/registration to get GCP Observability for gRPC.
-absl::Status GcpObservabilityInit();
+// GcpObservability objects follow the RAII idiom and help manage the lifetime
+// of gRPC Observability data exporting to GCP. `GcpObservability::Init()`
+// should be invoked instead to return an `GcpObservability` instance.
+// Observability data is flushed at regular intervals, and also when this
+// instance goes out of scope and its destructor is invoked.
+class GcpObservability {
+ public:
+  // Initialize GCP Observability for gRPC.
+  // This should be called before any other gRPC operations like creating a
+  // channel, server, credentials etc.
+  // The most common usage would call this at the top (or near the top) in
+  // main(). The return value helps determine whether observability was
+  // successfully enabled or not. On success, an object of class `Observability`
+  // is returned. When this object goes out of scope, GCP Observability stats,
+  // tracing and logging data is flushed. On failure, the status message can be
```

Comment:
Is it safe to do this (drop `GcpObservability`) whilst still making calls with gRPC?
What is the expected behavior?

Reply:
Current behavior - Just sleep for 25 sec to make sure current traces and stats are flushed. Logs are flushed and no more stuff is logged.

Possible future behavior with OpenTelemetry - Flush and stop exporting metrics for traces and stats. Same behavior for logs.

Reply:
I'll update the docs.